ts-force/src/rest/restHeader.ts

```diff
@@ -0,0 +1,30 @@
+export type SforceMru = {
+  updateMru: boolean;
+};
+export type PackageVersion = {
+  package: string;
+  version: string;
+};
+
+export type GenericHeader = {
+  header: string;
+  value: string;
+};
+
+export type RequestHeadersInput = {
+  'Sforce-Auto-Assign'?: boolean;
+  'Sforce-Mru'?: SforceMru;
+  'x-sfdc-packageversion'?: PackageVersion;
+  'If-Modified-Since'?: Date;
+  'If-Unmodified-Since'?: Date;
+  'If-Match'?: string[];
+  'If-None-Match'?: string[];
+  'Generic-Header'?: GenericHeader;
```

Comment:
Is `Generic-Header` just allowing an arbitrary header key/value?  

If so, it seems like you would want to potentially support multiple values.  It might also be easier to just allow a `Record<string, string>`.

`additionalHeaders?: Record<string, string>;`